docs/public/specifications/tests/Connectivity-TP/connectivity04.md

```diff
@@ -136,46 +137,56 @@ in England and Wales) and is continuously being mapped into
 <https://bgp.tools/table.txt>. The Cymru source can also be used, if
 requested.
 
+1. Input is the IP address in the call to this section.
 
-1. Prepend the *Cymru Base Name* with the label "origin" (IPv4) or 
+2. Prepend the *Cymru Base Name* with the label "origin" (IPv4) or
    "origin6" (IPv6) ("Expanded Base Name"). Example of expanded basenames :
    
 ```
 origin.asnlookup.zonemaster.net
 origin6.asnlookup.zonemaster.net
 ```
 
-2. Reverse the IP address with the same method as is used for
+3. Reverse the IP address in the input with the same method as is used for
    reverse lookup ("Reverse IP"). For description see [RFC 1035][RFC 1035#3.5],
    section 3.5, for IPv4 and [RFC 3596][RFC 3596#2.5], section 2.5, for IPv6.
  
-3. Prepend the *Expanded Base Name* with *Reverse IP* ("Query Name").
+4. Prepend the *Expanded Base Name* with *Reverse IP* ("Query Name").
    See [IP to ASN Mapping] for details.
 
-4. Create a [DNS Query] with query type TXT and query name *Query Name*.
+5. Create a [DNS Query] with query type TXT and query name *Query Name*.
    ("TXT Query").
 
-5. [Send] *TXT Query* to the *Cymru Base Name*.
+6. Do [DNS Lookup] of *TXT Query*.
 
-6. If at least one of the following criteria is met, output
+7. If at least one of the following criteria is met, output
    *[CN04_EMPTY_PREFIX_SET]* and exit this lookup:
    1. The [DNS Response] has the [RCODE Name] NXDomain.
    2. The [DNS Response] has the [RCODE Name] NoError and an empty answer section.
 
-7. If at least one of the following criteria is met, output
+8. If at least one of the following criteria is met, output
    *[CN04_ERROR_PREFIX_DATABASE]* and exit this lookup:
    1. There is no DNS response.
-   2. The [DNS Response] does not have the [RCODE Name] NoError or NXDomain.
+   2. The [DNS Response] does not have the [RCODE Name] NoError.
+   3. The answer section has no TXT record.
 
-8. Extract the TXT record(s) from the response (see [IP to ASN Mapping]
-   for examples), and do:
-   1. If there are multiple strings (i.e., TXT records), ignore all strings
-      except for the string with the most specific subnet.
-   2. Extract and [concatenate] the IP prefix from the string.
-   3. If it was not possible to extract the IP prefix (i.e., malformed response),
-      output *[CN04_ERROR_PREFIX_DATABASE]* and exit this lookup.
+9. Extract the TXT record(s) from the answer section (see [IP to ASN Mapping]
+   for examples). Do for each TXT record:
+   1. If the TXT record consists of multiple strings in RDATA, then [concatenate]
+      the strings into one string.
+   2. Using the format of such string parse the string into its parts and
```

Comment:
> the format of such string

Is there a specification for this we could/should refer to?

Reply:
No. My plan is to go backwards no to delay this change, i.e. to write that specification afterwards.